Add support for Protherm Panther 25KOV and Thermolink P (New)

WildRat opened the pull request:
Add support for:
Protherm Panther 25KOV _"MF=Vaillant;ID=BAI00;SW=0304;HW=4702"_
Thermolink P (New) _"MF=Vaillant;ID=E7C00;SW=0206;HW=7402"_
**All tested!**

samm-git commented:
I think we should merger your PR and one for the Thermolink rc/2, as they looks the same.

john30 commented on ebusd-2.1.x/en/vaillant/08.bai.csv:
```diff
@@ -25,5 +25,6 @@
 [HW=9602]!load,bai.0010015600.inc,,,
 [HW=8701]!load,bai.0020066007.inc,,,
 [HW=7401]!load,bai.308523.inc,,,
+[HW=4702]!load,bai.protherm25kov.inc,,,
```
use the product number instead of the name please (you can get it by using `ebusctl f -d -v -c scan.08 id`)

john30 reviewed (changes requested):
please also provide german version of the files

Trilis29 commented:
@WildRat Possible to control temperature flow without Thermolink???

WildRat commented:
Yes, you can set hot water temperature and heating water temperature:
r;w |   | FlowSetPotmeter | FlowSetPotmeter |   |   |   | F003 |   |   | temp |   |   | flow setpoint potentiometer
r;w |   | HwcSetPotmeter | DHWSetPotmeter |   |   |   | F103 |   |   | temp |   |   | DHW setpoint potentiometer
r;w |   | HeatingSwitch | Heating Switch |   |   |   | F203 |   |   | onoff |   |   | Wintermode active
r;w |   | HwcSwitch | DHWSwitch |   |   |   | F303 |   |   | onoff |   |   | DHW on/off

This is what you get on boiler screen. You can read and you can set this values as from boiler keyboard and it will work.

Trilis29 commented:
Can you give your telegram or other messanging? Want to consult on automation of Protherm

Trilis29 commented:
> Yes, you can set hot water temperature and heating water temperature:
> r;w |   | FlowSetPotmeter | FlowSetPotmeter |   |   |   | F003 |   |   | temp |   |   | flow setpoint potentiometer
> r;w |   | HwcSetPotmeter | DHWSetPotmeter |   |   |   | F103 |   |   | temp |   |   | DHW setpoint potentiometer
> r;w |   | HeatingSwitch | Heating Switch |   |   |   | F203 |   |   | onoff |   |   | Wintermode active
> r;w |   | HwcSwitch | DHWSwitch |   |   |   | F303 |   |   | onoff |   |   | DHW on/off
> 
> This is what you get on boiler screen. You can read and you can set this values as from boiler keyboard and it will work.

HeatingSwitch works. But if I change FlowSetPotmeter receiving the same value. As example read value 27, changed to 28 got answer "done". Reread again and still shows 27

samm-git commented:
@Trilis29 are you using radio module or directly connected one? Radio is resetting values from one in remote unit, and i did not found reliable way to override it.

WildRat commented:
It's the method of controlling without any thermostat, if you have thermostat you must set values in it. Also it depends on mode (heating, hot water, both or nether). Can you change values by keys, did they stay or reset?

Trilis29 commented:
> It's the method of controlling without any thermostat, if you have thermostat you must set values in it. Also it depends on mode (heating, hot water, both or nether). Can you change values by keys, did they stay or reset?

I will try later today. To check on all modes and with keys.
My plan is to put thermo valves on collector with relays connected to it, to control every room. I will get temperature of BLE or DS18B20 temperature meters and want to make automation. 

Easy way would be to leave flow temp  like 27. Slowly close water valves in collector which reached temperature, after reaching wanted temperature in all rooms shut off HeatingSwitch (or just add simple relay to Protherm and use this one to start/stop)

If you have any better advice, please let me know :) Would be nice to chat over telegra/signal/whatsapp



> @Trilis29 are you using radio module or directly connected one? Radio is resetting values from one in remote unit, and i did not found reliable way to override it.

using ebus-adapter 3 wemos now. Do not have spare RPI.

archont00 commented:
The file `bai.protherm25kov.inc` works well also with Protherm 30 KKO:
```
address 08: slave #11, scanned "MF=Vaillant;ID=BAI00;SW=0502;HW=3302"
scan.08 id = prefix=21;year=14;week=40;product=0010008863;supplier=1610;counter=005132;suffix=N4
```

moskovskiy82 commented:
Tried to use this for protherm ray 14.
Just downloaded the file and included into the 08.bai.

But get the following error. Not very informative to start with but...

2022-01-09 18:38:22.658 [update notice] sent scan-read scan.08  QQ=31: Vaillant;BAI00;0108;7503
2022-01-09 18:38:22.658 [bus notice] scan 08: ;Vaillant;BAI00;0108;7503
2022-01-09 18:38:22.666 [main notice] read scan config file vaillant/08.bai.csv for ID "bai00", SW0108, HW7503
2022-01-09 18:38:22.667 [main error] error resolving conditions: ERR: element not found, last error: condition scan id: message not found, condition scan id: message not found, condition scan id: message not found, condition scan id: message not found, condition scan id: message not found, condition scan id: message not found, condition scan id: message not found, condition scan id: message not found, condition scan id: message not found, condition scan id: message not found, condition scan id: message not found, condition scan id: message not found, condition scan id: message not found, condition scan id: message not found, condition scan id: message not found, condition scan id: message not found
2022-01-09 18:38:22.672 [main error] error executing instructions: ERR: element not found, last error: error resolving condition for "08.bai": ERR: element not found condition scan id: message not founderror resolving condition for "08.bai": ERR: element not found condition scan id: message not founderror resolving condition for "08.bai": ERR: element not found condition scan id: message not founderror resolving condition for "08.bai": ERR: element not found condition scan id: message not founderror resolving condition for "08.bai": ERR: element not found condition scan id: message not founderror resolving condition for "08.bai": ERR: element not found condition scan id: message not founderror resolving condition for "08.bai": ERR: element not found condition scan id: message not founderror resolving condition for "08.bai": ERR: element not found condition scan id: message not founderror resolving condition for "08.bai": ERR: element not found condition scan id: message not founderror resolving condition for "08.bai": ERR: element not found condition scan id: message not founderror resolving condition for "08.bai": ERR: element not found condition scan id: message not founderror resolving condition for "08.bai": ERR: element not found condition scan id: message not founderror resolving condition for "08.bai": ERR: element not found condition scan id: message not founderror resolving condition for "08.bai": ERR: element not found condition scan id: message not founderror resolving condition for "08.bai": ERR: element not found condition scan id: message not found, error loading "vaillant/bai.prothermray.inc" for "08": ERR: element not found vaillant/bai.prothermray.inc:9: ERR: element not found, field type POWER in field 0, error including "vaillant/hcmode.inc" for "08": ERR: element not found vaillant/hcmode.inc:4: ERR: element not found, field type DCFSTATE in field 0

moskovskiy82 commented:
Tried this on protherm skat. Added the following ot 08.bai
[HW=7503]!load,bai.prothermray.inc,,,

But get the following error. Not too much informative but maybe an idea where to start?

2022-01-09 19:22:32.718 [bus notice] scan 08: ;Vaillant;BAI00;0108;7503
2022-01-09 19:22:32.726 [main error] error reading templates in vaillant: ERR: element not found, last error: vaillant/_templates.csv:46: ERR: element not found, field type TEMP in field 0
2022-01-09 19:22:32.731 [main notice] read scan config file vaillant/08.bai.csv for ID "bai00", SW0108, HW7503
2022-01-09 19:22:32.732 [main error] error resolving conditions: ERR: element not found, last error: condition scan id: message not found, condition scan id: message not found, condition scan id: message not found, condition scan id: message not found, condition scan id: message not found, condition scan id: message not found, condition scan id: message not found, condition scan id: message not found, condition scan id: message not found, condition scan id: message not found, condition scan id: message not found, condition scan id: message not found, condition scan id: message not found, condition scan id: message not found, condition scan id: message not found, condition scan id: message not found
2022-01-09 19:22:32.737 [main error] error executing instructions: ERR: element not found, last error: error resolving condition for "08.bai": ERR: element not found condition scan id: message not founderror resolving condition for "08.bai": ERR: element not found condition scan id: message not founderror resolving condition for "08.bai": ERR: element not found condition scan id: message not founderror resolving condition for "08.bai": ERR: element not found condition scan id: message not founderror resolving condition for "08.bai": ERR: element not found condition scan id: message not founderror resolving condition for "08.bai": ERR: element not found condition scan id: message not founderror resolving condition for "08.bai": ERR: element not found condition scan id: message not founderror resolving condition for "08.bai": ERR: element not found condition scan id: message not founderror resolving condition for "08.bai": ERR: element not found condition scan id: message not founderror resolving condition for "08.bai": ERR: element not found condition scan id: message not founderror resolving condition for "08.bai": ERR: element not found condition scan id: message not founderror resolving condition for "08.bai": ERR: element not found condition scan id: message not founderror resolving condition for "08.bai": ERR: element not found condition scan id: message not founderror resolving condition for "08.bai": ERR: element not found condition scan id: message not founderror resolving condition for "08.bai": ERR: element not found condition scan id: message not found, error loading "vaillant/bai.prothermray.inc" for "08": ERR: element not found vaillant/bai.prothermray.inc:9: ERR: element not found, field type POWER in field 0, error including "vaillant/hcmode.inc" for "08": ERR: element not found vaillant/hcmode.inc:4: ERR: element not found, field type DCFSTATE in field 0
2022-01-09 19:22:32.737 [main notice] found messages: 2 (0 conditional on 26 conditions, 0 poll, 0 update)

Trilis29 commented:
Have you tried to run with defaul config files?

moskovskiy82 commented:
It was my mistake.
I can confirm that this config also works quite well with Protherm Skat Ray heater

scan.08 id = prefix=21;year=20;week=38;product=0010023648;supplier=3100;counter=013480;suffix=N1

WildRat commented:
Made the necessary changes: Added translation into German, renamed the files according to the equipment identifier. Also remove everything related to Thermolink P from this commit, since this issue is closed by [commit](https://github.com/john30/ebusd-configuration/pull/232). Everything seems to be ready to merge.

WildRat commented:
Additionally, based on discussion and checks by other users, added "Protherm 30 KKO" (scan.08 id = prefix=21;year=14;week=40;product=0010008863;supplier=1610;counter=005132;suffix=N4) and "Protherm Skat Ray" (scan.08 id = prefix=21;year=20;week=38;product=0010023648;supplier=3100;counter=013480;suffix=N1) to the list of devices supported by this configuration file.
